app/javascript/components/organisms/StatusHistoryPopover.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+
+import Icon from '~/components/atoms/Icon'
+import Popover from '~/components/molecules/Popover'
+import StatusItems from '~/components/molecules/lease_up_sidebar/StatusItems'
+import Loading from '../molecules/Loading'
+import apiService from '~/apiService'
+
+const MAX_UPDATES_TO_SHOW_DEFAULT = 4
+const StatusHistoryPopover = ({ applicationId }) => {
+  const [showingAllStatuses, setShowingAllStatuses] = useState(false)
+  const onShowHideStatusesToggled = () => setShowingAllStatuses(!showingAllStatuses)
+
+  const StatusIconButton = ({ onClick, ref }) => (
+    <button ref={ref} onClick={onClick} className='tiny icon margin-left--half'>
+      <Icon icon='list-unordered' size='medium' />
+    </button>
+  )
+  const [data, setData] = useState({ statusItems: [], loading: false })
+
+  const fetchStatusHistory = async () => {
+    try {
+      setData({ statusItems: data.statusItems, loading: true })
+      const items = await apiService.getFieldUpdateComments(applicationId)
+      setData({ statusItems: items, loading: false })
+    } catch (e) {
+      setData({ statusItems: data.statusItems, loading: false })
+    }
+  }
+
+  const onClick = async () => {
+    data.statusItems.length === 0 && (await fetchStatusHistory())
+  }
+
+  const numberOfStatusesToDisplay = showingAllStatuses
+    ? data.statusItems.length
+    : MAX_UPDATES_TO_SHOW_DEFAULT
+
+  return (
+    <Popover buttonElement={StatusIconButton} onButtonClick={onClick}>
+      {data.loading ? (
+        <Loading isLoading />
```

Comment:
I think with loading you can just wrap the whole thing in a loading component, and set isLoading conditionally
```
<Loading isLoading={data.isLoading}>
 <StatusItems ... />
</Loading>
```

Reply:
saw this in your PR this morning. will try it